runtime/pkg/duration/duration.go

```diff
@@ -104,3 +133,15 @@ func (d Duration) Sub(t time.Time) time.Time {
 	td := time.Duration(d.Second)*time.Second + time.Duration(d.Minute)*time.Minute + time.Duration(d.Hour)*time.Hour
 	return t.Add(-td)
 }
+
+type ToDateDuration struct {
+	anchor runtimev1.TimeGrain
+}
+
+func (d ToDateDuration) Add(t time.Time) time.Time {
+	return time.Time{}
+}
+
+func (d ToDateDuration) Sub(t time.Time) time.Time {
+	return TruncateTime(t, d.anchor, t.Location(), 1, 1) // TODO: get first day and month
```

Comment:
Does this TODO need to be addressed now?

Reply:
It is not really supported in the frontend. We can revisit this once we have scheduled reports released.